modules/apps/27-interchain-accounts/controller/keeper/handshake.go

```diff
@@ -104,8 +104,9 @@ func (k Keeper) OnChanCloseConfirm(
 
 // validateControllerPortParams asserts the provided connection sequence and counterparty connection sequence
 // match that of the associated connection stored in state
-func (k Keeper) validateControllerPortParams(ctx sdk.Context, channelID, portID string, connectionSeq, counterpartyConnectionSeq uint64) error {
-	connectionID, connection, err := k.channelKeeper.GetChannelConnection(ctx, portID, channelID)
+func (k Keeper) validateControllerPortParams(ctx sdk.Context, connectionHops []string, connectionSeq, counterpartyConnectionSeq uint64) error {
```

Comment:
applications must not rely on channel state being set before the application callback

Reply:
should we document this somewhere? :thinking:

Reply:
I'll update the docs when I do the OnChanOpenTry change

Reply:
Cool. I was just thinking out loud :)